### PR TITLE
Add feature to copy a contact's information to clipboard

### PR DESCRIPTION
**Pull request by jetrz**
Add "clip" function to allow users to efficiently copy a contact's information onto their system's clipboard with a simple command.

Add/update classes to facilitate this feature, including:
* Predicate class to search for exact Name String matching (as compared to sub-string matching used in "find" command)
* Function to compare Name String which ignores casing
* Parser class for parsing of this command
* Unit tests for new classes created

Add documentation for new feature.

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -207,6 +223,7 @@ Format: `exit`
  | **Delete** | `delete INDEX`<br> e.g., `delete 3`|
 | **Edit**   | `edit INDEX [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [a/ADDRESS] [i/INSURANCE_PACKAGE] [t/TAG]…​`<br> e.g.,`edit 2 n/James Lee e/jameslee@example.com`|
  | **Find**   | `find FIELD KEYWORD [MORE_KEYWORDS]`<br> e.g., `find n/James Jake`|
+ | **Clip**    | `clip n/NAME`<br> e.g., `find n/John Doe`|
```
Comment (leeyiheng12) [inline docs/UserGuide.md:226]: Did you mean "e.g., `clip n/John Doe`"?

Comment (leeyiheng12) [reply docs/UserGuide.md:226]: Similar to how after a list or find, `delete INDEX` deletes the details at that specified INDEX, perhaps `clip INDEX` can do something similar?

File: src/test/java/seedu/address/logic/commands/AddToClipboardCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+     * Test for when a person is found matching the given keyword.
+     * Should return the success message from AddToClipboardCommand.java class.
+     */
+    @Test
+    void execute_personFound() throws HeadlessException {
+        String expectedMessage = AddToClipboardCommand.MESSAGE_SUCCESS;
+        NameExistsPredicate predicate = preparePredicate("Elle Meyer");
+        AddToClipboardCommand command = new AddToClipboardCommand(predicate);
+        try {
+            //This has to be implemented as automated testing on github does not have a "clipboard" to copy to,
```
Comment (leeyiheng12) [inline src/test/java/seedu/address/logic/commands/AddToClipboardCommandTest.java:91]: Good catch!

**Review by leeyiheng12 — APPROVED** [review]
Everything looks good, nice work! This is a great feature to have.

File: docs/UserGuide.md (hunk 1 above)
Comment (jetrz) [reply docs/UserGuide.md:226]: That definitely sounds like a good enhancement for the clip command! Great idea!

**Pull request merged**